Mathlib/AlgebraicGeometry/EllipticCurve/NormalForms.lean

```diff
@@ -101,10 +100,10 @@ variable {K : Type*} [Field K] (W₁ : WeierstrassCurve K) (E : EllipticCurve K)
 
 namespace WeierstrassCurve
 
-/-! ### Normal forms of characteristic different from two -/
+/-! ### Normal forms of characteristic ≠ 2 -/
 
-/-- A `WeierstrassCurve` is in normal form of characteristic ≠ 2, if its $a_1$ and $a_3$
-coefficients are zero. In other words it is $Y^2 = X^3 + a_2X^2 + a_4X + a_6$. -/
+/-- A `WeierstrassCurve` is in normal form of characteristic ≠ 2, if its $a_1, a_3 = 0$.
+In other words it is $Y^2 = X^3 + a_2X^2 + a_4X + a_6$. -/
 @[mk_iff]
 structure IsCharNeTwoNF : Prop where
   a₁ : W.a₁ = 0
```

Comment:
A more genuine question: is there a point in using `structure` rather than just making this a `def`? You have five instances of `structure` and two instances of `def` for `IsChar.*`, and we should probably make this consistent.

I personally prefer `def` since we don't need `_spec`, but maybe there's a reason you need `structure`.

Reply:
> A more genuine question: is there a point in using `structure` rather than just making this a `def`?

Of course, for example I can use `def IsCharNeTwoThreeNF := W.a₁ = 0 ∧ W.a₂ = 0 ∧ W.a₃ = 0`. But then it's not straightforward to reference `W.a₁ = 0`.

> You have five instances of `structure` and two instances of `def` for `IsChar.*`, and we should probably make this consistent.

Do you mean `inductive IsCharThreeNF : Prop` instead of `def IsCharThreeNF : Prop := W.IsCharThreeJNeZeroNF ∨ W.IsCharNeTwoThreeNF`? Not that this condition is `or`, not `and`. But I think `inductive` version does not give us advantages.

> I personally prefer `def` since we don't need `_spec`, but maybe there's a reason you need `structure`.

I don't get your point. The `_spec` is for the property of the variable change `toCharNeTwoNF`, but not for the prop `IsCharNeTwoNF` itself.

Reply:
Ah fair enough, this was my mistake because I was scrolling too quickly.